maker/src/contract_setup.rs

```diff
@@ -91,19 +92,26 @@ impl Actor {
             .await?;
 
         let taker_id = setup_params.counterparty_identity();
+        let taker = self.taker.clone();
 
         let contract_future = setup_contract_deprecated::new(
-            self.taker.sink().with(move |msg| {
-                future::ok(connection::TakerMessage {
-                    taker_id,
-                    msg: wire::MakerToTaker::Protocol { order_id, msg },
-                })
+            sink::unfold((), move |_, msg| {
+                let taker = taker.clone();
+                async move {
+                    let msg = connection::TakerMessage {
+                        taker_id,
+                        msg: wire::MakerToTaker::Protocol { order_id, msg },
+                    };
+
+                    let _ = taker.send(msg).split_receiver().await;
```

Comment:
if we can easily log the error here, I'd opt for that (as it is an improvement :) ). Perhaps something like:

```suggestion
                    if let Err(e) = taker.send(msg).split_receiver().await {
                        tracing::error!("failed to send contract_setup message to the taker: {e:#});
                    } 
```

(in all instances where you introduced unfolding)

Reply:
Can do!

Reply:
Ah, we cannot actually do this currently without https://github.com/Restioson/xtra/pull/105, unless we first wait for the message to be handled and then print the result. This has a different backpressure behaviour

Reply:
Additionally, should the sink stop accepting messages if an error occurs? I think this can be done by returning `Err` from `unfold`

Reply:
interesting idea, although I would probably not go that far yet, as this will indeed change the behaviour. I would like to first get some stats on how often we see errors here.

Reply:
> Ah, we cannot actually do this currently without [Restioson/xtra#105](https://github.com/Restioson/xtra/pull/105), unless we first wait for the message to be handled and then print the result. This has a different backpressure behaviour. On the other hand, it could still log an error if the taker is disconnected right after sending the message, or right before.

Ok, no worries! we can also print the messages later. happy for it to go without printing these logs :)